bundles/paikkatietoikkuna/mapmodule/plugin/LayersPlugin3D.olcs.js

```diff
@@ -0,0 +1,64 @@
+
+const LayersPluginOl3Impl = Oskari.clazz.get('Oskari.mapframework.bundle.mapmodule.plugin.LayersPlugin');
+
+class LayersPluginOlcsImpl extends LayersPluginOl3Impl {
+    /**
+     * @method _isLayerImplVisible Tells if given layer is visible or not. 
+     * @param {Cesium.Cesium3DTileset | olLayer} layer 
+     */
+    _isLayerImplVisible (layer) {
+        if (layer instanceof Cesium.Cesium3DTileset) {
+            return layer.show;
+        } else {
+            return layer.getVisible();
+        }
+    }
+    /**
+     * @method _setLayerImplVisible Sets layer visibility on map.
+     * @param {Cesium.Cesium3DTileset | olLayer} layer 
+     * @param {Boolean} visible 
+     */
+    _setLayerImplVisible (layer, visible) {
+        if (layer instanceof Cesium.Cesium3DTileset) {
+            layer.show = visible;
+        } else {
+            return layer.setVisible(visible);
+        }
+    }
+    /**
+     * @method handleMapLayerVisibility
+     * Checks layer's visibility (visible, inScale, inGeometry) and sets ol layers' visibilities.
+     * notifies bundles about visibility changes by sending MapLayerVisibilityChangedEvent.
+     * @param {Oskari.mapframework.domain.AbstractMapLayer} layer layer to check against
+     * @param {Boolean} isRequest if MapLayerVisibilityRequest, then trigger always change because layer's visibility has changed
+     */
+    handleMapLayerVisibility (layer, isRequest) {
+        let scaleOk = layer.isVisible();
+        let geometryMatch = layer.isVisible();
+        let triggerChange = (isRequest === true);
+        // if layer is visible check actual values
+        if (layer.isVisible()) {
+            scaleOk = this._isInScale(layer);
+            geometryMatch = this.isInGeometry(layer);
+        }
+        const mapLayers = this.getMapModule().getOLMapLayers(layer.getId());
+        if (!mapLayers || !mapLayers.length) {
+            if (triggerChange) {
+                this.notifyLayerVisibilityChanged(layer, scaleOk, geometryMatch);
+            }
+            return;
+        }
+        const visibility = scaleOk && geometryMatch && layer.isVisible();
+        mapLayers.forEach(mapLayer => {
+            if (visibility !== this._isLayerImplVisible(mapLayer)) {
+                this._setLayerImplVisible(mapLayer, visibility);
+                triggerChange = true;
+            }
+        });
+        if (triggerChange) {
+            this.notifyLayerVisibilityChanged(layer, scaleOk, geometryMatch);
+        }
+    }
+}
+
+Oskari.clazz.defineES('Oskari.mapframework.bundle.mapmodule.plugin.LayersPlugin3D', LayersPluginOlcsImpl);
```

Comment:
It's better to have the same name for the same functionality even if the implementation changes. Let's refactor this the same way as ol <> olcs mapmodule share the same base mapmodule.